client/src/ui/organisms/sidebar/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+import "./index.scss";
+
+export function RenderSideBar({ doc }) {
+  console.log(doc);
+  if (!doc.related_content) {
+    if (doc.sidebarHTML) {
+      return (
+        <aside
+          id="sidebar-quicklinks"
+          className="sidebar"
+          dangerouslySetInnerHTML={{
+            __html: `<h4>Related Topics</h4>${doc.sidebarHTML}`,
+          }}
+        />
+      );
+    }
+    return null;
+  }
+  return doc.related_content.map((node) => (
+    <SidebarLeaf key={node.title} parent={node} />
+  ));
+}
+
+function SidebarLeaf({ parent }) {
+  return (
+    <aside id="sidebar-quicklinks" className="sidebar">
+      <h4>{parent.title}</h4>
+      <ul>
+        {parent.content.map((node) => {
+          if (node.content) {
+            return (
+              <li key={node.title}>
+                <SidebarLeaflets node={node} />
+              </li>
+            );
+          } else {
+            return (
+              <li key={node.uri}>
+                <Link to={node.uri}>{node.title}</Link>
+              </li>
+            );
+          }
+        })}
+      </ul>
+    </aside>
+  );
+}
+
+function SidebarLeaflets({ node }) {
+  return (
+    <details open={node.open}>
+      <summary>
+        {node.uri ? <Link to={node.uri}>{node.title}</Link> : node.title}
+      </summary>
+      <ol>
+        {node.content.map((childNode) => {
+          if (childNode.content) {
+            return (
+              <li key={childNode.title}>
+                <SidebarLeaflets node={childNode} />
+              </li>
+            );
+          } else {
+            return (
+              <li
+                key={childNode.uri}
+                className={childNode.isActive && "active"}
+              >
+                <Link to={childNode.uri}>{childNode.title}</Link>
+              </li>
+            );
+          }
+        })}
+      </ol>
+    </details>
+  );
+}
```

Comment:
I know this was mostly a cut-and-paste job but you did make changes on top of that. It would have been nice to know what those were so I didn't have to review it all wholesale.

Reply:
Good point. I will be sure to separate those out into separate commits in future. My apologies Peter.